fe/fe-core/src/main/java/org/apache/doris/load/loadv2/TokenManager.java

```
@@ -0,0 +1,133 @@
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.load.loadv2;
+
+import org.apache.doris.catalog.Env;
+import org.apache.doris.common.ClientPool;
+import org.apache.doris.common.Config;
+import org.apache.doris.common.FeConstants;
+import org.apache.doris.thrift.FrontendService;
+import org.apache.doris.thrift.TMySqlLoadAcquireTokenResult;
+import org.apache.doris.thrift.TNetworkAddress;
+import org.apache.doris.thrift.TStatusCode;
+
+import com.google.common.collect.EvictingQueue;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.thrift.TException;
+import org.apache.thrift.transport.TTransportException;
+
+import java.util.UUID;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+public class TokenManager {
+    private static final Logger LOG = LogManager.getLogger(TokenManager.class);
+
+    private final int thriftTimeoutMs = 300 * 1000;
+    private final EvictingQueue<String> tokenQueue;
+    private final ScheduledExecutorService tokenGenerator;
+
+    public TokenManager() {
+        this.tokenQueue = EvictingQueue.create(Config.token_queue_size);
+        // init one token to avoid async issue.
+        this.tokenQueue.offer(generateNewToken());
+        this.tokenGenerator = Executors.newScheduledThreadPool(1);
+        this.tokenGenerator.scheduleAtFixedRate(() -> {
+            tokenQueue.offer(generateNewToken());
+        }, 0, Config.token_generate_period_hour, TimeUnit.HOURS);
+    }
+
+    private String generateNewToken() {
+        return UUID.randomUUID().toString();
+    }
+
+    // this method only will be called in master node, since stream load only send message to master.
+    public boolean checkAuthToken(String token) {
+        return tokenQueue.contains(token);
+    }
+
+    public String acquireToken() {
```

Better throw exception when acquiring token failed, instead of returning `null`.
It is every error prone and I saw some place which not handle `null` value.